### ISSUE TITLE
Bigint support

### ISSUE THREAD

**Issue by protango** [MikeMcl/decimal.js#181]
With [bigint](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) now a part of JavaScript, it would be nice for Decimal.JS to support it within its constructor and operations.

As a workaround we can just convert bigint's to strings before using them in Decimal.JS, however this becomes very repetitive with lots of numbers.

**Comment by mahnunchik** [comment]
Any news?

**Comment by MikeMcl** [comment]
@mahnunchik 

No news. I'll try and have a look at this soon.

**Comment by sterlu** [comment]
Would you accept a PR which treats BigInt as a string in the constructor? Should be pretty simple to do.

**Comment by MikeMcl** [comment]
@sterlu 

Yes, but if the argument is a BigInt then there would be no need for the minus and plus sign checks, or for the `isDecimal.test(v)`, so it would just be a matter of setting `x.s` to `1` or `-1` depending on whether the argument is positive or negative, and then returning `parseDecimal(x, v)`. Ça va?

**Comment by SynthLuvr** [comment]
This would be good to add to make our codebase a little bit more concise. I think @sterlu is on the right track, there wouldn't be too much effort required to add bigint support.

**Comment by MikeMcl** [comment]
Added  `BigInt` support in *v10.5.0*.

**Comment by SynthLuvr** [comment]
> Added `BigInt` support in _v10.5.0_.

Type definition is missing: https://github.com/MikeMcl/decimal.js/blob/master/decimal.d.ts#L39

**Comment by MikeMcl** [comment]
@SynthLuvr 

Arghhh, always forget something!

Too late, for now.   

Still to consider #234 properly, so will get to it then.

Thanks for the heads up.

**Comment by posquit0** [comment]
@MikeMcl Can you release a new version with the patch (type definition).

**Comment by MikeMcl** [comment]
@sterlu 

Published in *v10.6.0*. Thanks for the nudge.
